neuraxle/metaopt/auto_ml.py

```diff
@@ -0,0 +1,358 @@
+                'hyperparams': hp_dict,
+                'score': score
+            }, outfile)
+
+
+class AutoMLStrategyMixin:
+    """
+    Base class for Automatic Machine Learning strategies.
+    Implement your own custom intelligent search of hyperparameters to get most accurate predictive models.
+
+    .. seealso::
+        :class:`BaseCrossValidation`,
+        :class:`HyperparameterSamples`,
+        :class:`HyperparameterSpace`,
+        :class:`MetaStepMixin`,
+        :class:`BaseStep`
+    """
+
+    def __init__(self, validation_technique: BaseCrossValidationWrapper = None, higher_score_is_better=True):
+        if validation_technique is None:
+            validation_technique = KFoldCrossValidationWrapper()
+        self.validation_technique = validation_technique
+        self.higher_score_is_better = higher_score_is_better
+
+    @abstractmethod
+    def find_next_best_hyperparams(self, trials_data_container: 'AutoMLDataContainer') -> HyperparameterSamples:
+        """
+        Find the next best hyperparams using previous trials.
+
+        :param trials_data_container: trials data container
+        :type trials_data_container: AutoMLDataContainer
+        :return: next best hyperparams
+        :rtype: HyperparameterSamples
+        """
+        raise NotImplementedError()
+
+    def fit_transform(self, data_inputs, expected_outputs=None) -> ('AutoMLStrategyMixin', float):
+        """
+        Fit cross validation with wrapped step, and return the score.
+
+        :param data_inputs: data inputs
+        :param expected_outputs: expected outputs to fit on
+        :return: self, step score
+        :rtype: (AutoMLStrategyMixin, float)
+        """
+        step = copy.copy(self.wrapped)
+        step: BaseCrossValidationWrapper = copy.copy(self.validation_technique).set_step(step)
+        step = step.fit(data_inputs, expected_outputs)
+
+        return self, step.get_score()
+
+    def is_higher_score_better(self):
+        return self.higher_score_is_better
+
+    def fit(self, data_inputs, expected_outputs=None):
+        return self
+
+    def transform(self, data_inputs, expected_outputs=None):
+        return data_inputs
+
+
+class RandomSearchBaseAutoMLStrategy(AutoMLStrategyMixin, MetaStepMixin, BaseStep):
+    """
+    Random Search Automatic Machine Learning Strategy that randomly samples the space of random variables.
+
+    .. seealso::
+        :class:`AutoMLStrategyMixin`,
+        :class:`AutoMLDataContainer`,
+        :class:`HyperparameterSamples`,
+        :class:`HyperparameterSpace`
+    """
+
+    def __init__(self, validation_technique=None):
+        AutoMLStrategyMixin.__init__(self, validation_technique)
+        MetaStepMixin.__init__(self, None)
+        BaseStep.__init__(self)
+
+    def find_next_best_hyperparams(self, trials_data_container: 'AutoMLDataContainer') -> HyperparameterSamples:
+        return trials_data_container.hyperparameter_space.rvs()
+
+
+class AutoMLDataContainer:
```

Comment:
I wouldn't have expected to see this class exist. I doesn't inherit from a data container so it isn't one.

Reply:
Should this class instead be named `Trials`? 

Is this class representing 1 or many trials? there is a mix of 1 items like `trial_number: int` and N-items like `scores: List[float]`. 

I feel something is wrong here, I can't tell exactly.

Reply:
well it is just a data object that is easier to play with then a thousand arguments

Reply:
the class contains the index because certain algorithm need it. but it contains multiple trials

Reply:
renamed to TrialsContainer